we don't need DESTDIR here.
and adjest destination dir to PREFIX/man/ from PREFIX/share/man/.

Index: doc/Makefile
--- doc/Makefile.orig
+++ doc/Makefile
@@ -1,5 +1,5 @@
 PREFIX ?= /usr/local
-MANDIR ?= $(PREFIX)/share/man
+MANDIR ?= $(PREFIX)/man
 INSTALL ?= install
 
 all: man/wyrd.1 man/wyrd.7 man/wyrdrc.5
@@ -81,17 +81,17 @@ man/wyrdrc.5: page-wyrdrc.odocl footer.man
 	rmdir man/manual || true
 
 install: all
-	$(INSTALL) -d $(DESTDIR)$(MANDIR)/man1 \
-		      $(DESTDIR)$(MANDIR)/man7 \
-		      $(DESTDIR)$(MANDIR)/man5
-	$(INSTALL) -m 644 man/wyrd.1   $(DESTDIR)$(MANDIR)/man1/
-	$(INSTALL) -m 644 man/wyrd.7   $(DESTDIR)$(MANDIR)/man7/
-	$(INSTALL) -m 644 man/wyrdrc.5 $(DESTDIR)$(MANDIR)/man5/
+	$(INSTALL) -d $(MANDIR)/man1 \
+		      $(MANDIR)/man7 \
+		      $(MANDIR)/man5
+	$(INSTALL) -m 644 man/wyrd.1   $(MANDIR)/man1/
+	$(INSTALL) -m 644 man/wyrd.7   $(MANDIR)/man7/
+	$(INSTALL) -m 644 man/wyrdrc.5 $(MANDIR)/man5/
 
 uninstall:
-	rm -f $(DESTDIR)$(MANDIR)/man1/wyrd.1 \
-	      $(DESTDIR)$(MANDIR)/man7/wyrd.7 \
-	      $(DESTDIR)$(MANDIR)/man5/wyrdrc.5
+	rm -f $(MANDIR)/man1/wyrd.1 \
+	      $(MANDIR)/man7/wyrd.7 \
+	      $(MANDIR)/man5/wyrdrc.5
 
 distclean:
 	latexmk -c || true
